tests/filesystem_test.py

```diff
@@ -90,6 +93,13 @@ def test_remote_path_suffix():
     assert RemotePath("source/file").suffix is None
 
 
+def test_remote_path_truncated():
+    assert RemotePath("something-long.txt").truncated(10) == "someth.txt"
+    assert RemotePath("longlonglong/short").truncated(5) == "longl/short"
+    assert RemotePath("a-long.data.dir/filename.csv").truncated(7) == "a-l.dir/fil.csv"
```

Comment:
Why does `a-long` become `a-l`, while `something-long` became `someth` above?

Reply:
Because the two are truncated to different lengths.